ui/src/features/WorkspaceSettings/components/MembersSettings.tsx

```diff
@@ -3,87 +3,109 @@ import { useState } from "react";
 
 import {
   Button,
-  Checkbox,
   DropdownMenu,
   DropdownMenuContent,
   DropdownMenuItem,
   DropdownMenuTrigger,
+  Input,
 } from "@flow/components";
+import { useUser, useWorkspace } from "@flow/lib/gql";
 import { useT } from "@flow/lib/i18n";
 import { useCurrentWorkspace } from "@flow/stores";
 import { Role, UserMember } from "@flow/types";
 
 type Filter = "all" | Role;
 
-const roles = ["admin", "reader", "writer"];
+const roles: Role[] = Object.values(Role);
 
 const MembersSettings: React.FC = () => {
   const t = useT();
-  const [currentWorkspace] = useCurrentWorkspace();
-
+  const [currentWorkspace, setCurrentWorkspace] = useCurrentWorkspace();
+  const { addMemberToWorkspace, removeMemberFromWorkspace, updateMemberOfWorkspace } =
+    useWorkspace();
+  const { searchUser } = useUser();
+  const [email, setEmail] = useState<string>("");
   const [currentFilter, setFilter] = useState<Filter>("all");
+  const [error, setError] = useState<string | undefined>();
 
   const filters: { id: Filter; title: string }[] = [
     { id: "all", title: t("All") },
-    { id: "admin", title: t("Admin") },
-    { id: "reader", title: t("Reader") },
-    { id: "writer", title: t("Writer") },
+    { id: Role.Owner, title: t("Owner") },
+    { id: Role.Reader, title: t("Reader") },
+    { id: Role.Reader, title: t("Maintainer") },
+    { id: Role.Writer, title: t("Writer") },
   ];
 
   const members =
-    (currentWorkspace?.members?.filter(
-      m => "userId" in m && (currentFilter !== "all" ? m.role === currentFilter : true),
-    ) as UserMember[]) ?? [];
+    (
+      currentWorkspace?.members?.filter(
+        m => "userId" in m && (currentFilter !== "all" ? m.role === currentFilter : true),
```

Comment:
```suggestion
        m => "userId" in m && (currentFilter === "all" || m.role === currentFilter),
```

Reply:
Oh yea, sorry I changed it...i was checking something.